test/unit/controllers/consolidations/AssociateDucrSummaryControllerSpec.scala

```diff
@@ -73,131 +62,102 @@ class AssociateDucrSummaryControllerSpec extends ControllerSpec with MockSubmiss
+        withCaching(AssociateDucr.formId, None)
 
-      "call SubmissionService and CustomsCacheService in correct order" in {
-
-        when(mockSubmissionService.submitDucrAssociation(any(), any())(any(), any()))
-          .thenReturn(Future.successful(ACCEPTED))
-        when(mockCustomsCacheService.remove(anyString())(any(), any()))
-          .thenReturn(Future.successful(mock[HttpResponse]))
-
-        controller.submit()(postRequest(Json.obj())).futureValue
-
-        val inOrder: InOrder = Mockito.inOrder(mockSubmissionService, mockCustomsCacheService)
-        inOrder
-          .verify(mockCustomsCacheService)
-          .fetchAndGetEntry[MucrOptions](any(), meq(MucrOptions.formId))(any(), any(), any())
-        inOrder
-          .verify(mockCustomsCacheService)
-          .fetchAndGetEntry[AssociateDucr](any(), meq(AssociateDucr.formId))(any(), any(), any())
-        inOrder
-          .verify(mockSubmissionService)
-          .submitDucrAssociation(meq(MucrOptions("MUCR")), meq(AssociateDucr("DUCR")))(any(), any())
-        inOrder.verify(mockCustomsCacheService).remove(anyString)(any(), any())
+        assertThrows[IncompleteApplication] {
+          await(controller.displayPage()(getRequest()))
+        }
       }
 
-      "Redirect to next page" in {
-
-        when(mockSubmissionService.submitDucrAssociation(any(), any())(any(), any()))
-          .thenReturn(Future.successful(ACCEPTED))
-        when(mockCustomsCacheService.remove(anyString())(any(), any()))
-          .thenReturn(Future.successful(mock[HttpResponse]))
+      "Mucr Options is missing during submitting page" in {
 
-        val result = controller.submit()(postRequest(Json.obj()))
+        withCaching(MucrOptions.formId, None)
 
-        redirectLocation(result) mustBe Some(routes.AssociateDucrConfirmationController.displayPage().url)
+        assertThrows[IncompleteApplication] {
+          await(controller.submit()(postRequest(Json.obj())))
+        }
       }
 
-      "add MUCR to Flash" in {
-
-        when(mockSubmissionService.submitDucrAssociation(any(), any())(any(), any()))
-          .thenReturn(Future.successful(ACCEPTED))
-        when(mockCustomsCacheService.remove(anyString())(any(), any()))
-          .thenReturn(Future.successful(mock[HttpResponse]))
+      "Associate Ducr is missing during submitting page" in {
 
-        val result = controller.submit()(postRequest(Json.obj()))
+        withCaching(MucrOptions.formId, Some(MucrOptions("MUCR")))
+        withCaching(AssociateDucr.formId, None)
 
-        flash(result).get(FlashKeys.MUCR) mustBe Some("MUCR")
+        assertThrows[IncompleteApplication] {
+          await(controller.submit()(postRequest(Json.obj())))
+        }
       }
-
     }
 
-    "SubmissionService returns status other than Accepted" should {
+    "return 303 (SEE_OTHER)" when {
 
-      "return InternalServerError code" in {
+      "submission service returned ACCEPTED and all mandatory data is in cache" in {
```

Comment:
Nit: I would swap conditions. You have to have data in cache first, then you can think about having ACCEPTED returned.

Reply:
fixed